docs/source/algorithms/ReflectometryILLPreprocess-v1.rst

```diff
@@ -25,103 +24,47 @@ The algorithm can be thought as an 'advanced loader', and should be used to load
-###############
+.. diagram:: ReflectometryILLPreprocessReflected-v1_wkflw.dot
 
-A fitting of the present peak position takes place in order to determine the detector angles.
-For preventing fitting of the present peak position, the property *LinePosition* allows to provide a peak position.
-A use case is to enter a direct peak position, which can be obtained from the direct beam workspaces sample logs, when the *Run* is a reflected beam.
+Measurement
+###########
 
-Alternatively, the properties *TwoTheta* and the sample log *reduction.line_position* affect the pixel :math:`2\theta` angle.
+This tag defines whether the runs need to be processed as direct or reflected beams.
+In both cases the fractional foreground centre is fitted.
+For direct beam, the detector is rotated such that the foreground centre corresponds exactly to zero scattering angle.
+For reflected beam there are three possibilities, depending on the **AngleOption**:
+
+#. **UserAngle**: the detector is rotated such that the foreground centre of the reflected beam corresponds exactly to **2*BraggAngle**.
+#. **SampleAngle**: the detector is rotated such that the foreground centre if the reflected beam corresponds exactly to **2*SAN**, where the SAN is read from the nexus file of the reflected beam run.
+#. **DetectorAngle**: the detector is rotated such that the foreground centre of the corresponding direct beam corresponds exactly to **DAN_RB - DAN_DB**, where the two detector angles are read from the direct and reflected beams correspondingly. In this case the **DirectBeamDetectorAngle** and **DirectBeamForegroundCentre** are required, which could be retrieved from the sample logs of the pre-processed direct beam runs.
 
 Foreground and backgrounds
 ##########################
 
 Foreground is a set of pixels intensities of which will be summed in :ref:`ReflectometryILLSumForeground <algm-ReflectometryILLSumForeground>`. However, foreground needs to be defined already in this algorithm as the information is needed for the background pixels. The foreground pixel information is stored in the sample logs of *OutputWorkspace* under the entries starting with ``foreground.``.
 
-Background, on the other hand, is a set of pixels which are be used for fitting a constant or linear background by :ref:`CalculatePolynomialBackground <algm-CalculatePolynomialBackground>`.
-
-The foreground pixels are defined by the foreground centre and *ForegroundHalfWidth* property. In normal use cases, the foreground center (workspace index) is taken from the fitting in :ref:`LoadILLReflectometry <algm-LoadILLReflectometry>`. This can be overridden by giving the pixel as *BeamCentre*. Fractional values are rounded to nearest integer. The full process of deciding the foreground centre is as follows:
+Background, on the other hand, is a set of pixels which are be used for average (default), fitted constant or linear background by :ref:`CalculatePolynomialBackground <algm-CalculatePolynomialBackground>`.
```

Comment:
Should the average background calculation method be mentioned here?

Reply:
Yes, that's what I refer to as default in parentheses, perhaps could have been formulated better.

Reply:
Ah yes, it reads fine this morning. Must just have been tired last night!